packages/contracts-bedrock/scripts/L2Genesis.s.sol

```diff
@@ -0,0 +1,315 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Script } from "forge-std/Script.sol";
+import { console2 as console } from "forge-std/console2.sol";
+
+import { Artifacts } from "scripts/Artifacts.s.sol";
+import { DeployConfig } from "scripts/DeployConfig.s.sol";
+import { Predeploys } from "src/libraries/Predeploys.sol";
+import { L1StandardBridge } from "src/L1/L1StandardBridge.sol";
+import { L1CrossDomainMessenger } from "src/L1/L1CrossDomainMessenger.sol";
+import { L2StandardBridge } from "src/L2/L2StandardBridge.sol";
+import { L2CrossDomainMessenger } from "src/L2/L2CrossDomainMessenger.sol";
+import { SequencerFeeVault } from "src/L2/SequencerFeeVault.sol";
+import { FeeVault } from "src/universal/FeeVault.sol";
+import { OptimismMintableERC20Factory } from "src/universal/OptimismMintableERC20Factory.sol";
+import { L1Block } from "src/L2/L1Block.sol";
+import { GovernanceToken } from "src/governance/GovernanceToken.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+
+contract L2Genesis is Script, Artifacts {
+    uint256 constant PROXY_COUNT = 2048;
+    uint256 constant PRECOMPILE_COUNT = 256;
+    DeployConfig public constant cfg =
+        DeployConfig(address(uint160(uint256(keccak256(abi.encode("optimism.deployconfig"))))));
+
+    /// @notice The storage slot that holds the address of a proxy implementation.
+    /// @dev `bytes32(uint256(keccak256('eip1967.proxy.implementation')) - 1)`
+    bytes32 internal constant PROXY_IMPLEMENTATION_ADDRESS =
+        0x360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc;
+
+    /// @notice The storage slot that holds the address of the owner.
+    /// @dev `bytes32(uint256(keccak256('eip1967.proxy.admin')) - 1)`
+    bytes32 internal constant PROXY_ADMIN_ADDRESS = 0xb53127684a568b3173ae13b9f8a6016e243e63b6e8ee1178d6a717850b5d6103;
+
+    string internal outfile;
+    address[] internal addrToRemoveFromAlloc;
+
+    function setUp() public override {
+        Artifacts.setUp();
+
+        string memory path = string.concat(vm.projectRoot(), "/deploy-config/", deploymentContext, ".json");
+        vm.etch(address(cfg), vm.getDeployedCode("DeployConfig.s.sol:DeployConfig"));
+        vm.label(address(cfg), "DeployConfig");
+        vm.allowCheatcodes(address(cfg));
+        cfg.read(path);
+
+        addrToRemoveFromAlloc.push(address(cfg));
+
+        outfile = string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/genesis.json");
+
+        _loadAddresses(string.concat(vm.projectRoot(), "/deployments/", deploymentContext, "/.deploy"));
+    }
+
+    function run() public {
+        _setPrecompiles();
+        _setProxies();
+        _setImplementations();
+
+        vm.dumpState(outfile);
+
+        for (uint256 i = 0; i < addrToRemoveFromAlloc.length; i++) {
+            _removeAllocFromGenesisFile(outfile, addrToRemoveFromAlloc[i]);
+        }
```

Comment:
@tynes Previously we were using

```solidity
vm.etch(address(bridge), hex"");
vm.resetNonce(address(bridge));
```

to reset the account when we deployed the contract to obtain the deployed bytecode, but this didn't remove storage slots from the genesis `alloc` resulting in something like:

```json
"0x2e234dae75c793f67a35089c9d99245e1c58470b": {
    "nonce": 0,
    "balance": "0x0",
    "code": "0x",
    "storage": {
      "0x0000000000000000000000000000000000000000000000000000000000000003": "0x4f7074696d69736d000000000000000000000000000000000000000000000010",
      "0x0000000000000000000000000000000000000000000000000000000000000004": "0x4f50000000000000000000000000000000000000000000000000000000000004",
      "0x000000000000000000000000000000000000000000000000000000000000000a": "0x0000000000000000000000007fa9385be102ac3eac297483dd6233d62b3e1496"
    }
  },
```

I've added `_removeAllocFromGenesisFile` to completely remove the `alloc` object using an address